src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -106,8 +123,11 @@ trait CachingDisabled extends CachingEnabled {
   override def cachingEnabled: Boolean = false
 }
 
-class PersistedProducer[T, -A <: Artifact](step: Producer[T], io: SerializeToArtifact[T, A] with DeserializeFromArtifact[T, A],
-  _artifact: A) extends Producer[T] {
+class PersistedProducer[T, -A <: Artifact](
```

Comment:
Do we want logic in `PersistedProducer` to return `timeTaken = None` if the input artifact exists?